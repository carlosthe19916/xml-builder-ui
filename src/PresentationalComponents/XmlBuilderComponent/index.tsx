import XmlBuilderComponent from "./XmlBuilderComponent";

export default XmlBuilderComponent;
