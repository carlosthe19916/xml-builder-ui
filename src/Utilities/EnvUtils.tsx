export const ENV_SIGNER_MODE = "REACT_APP_SIGNER_MODE";

export const isAppInSignerMode = () => {
  return process.env[ENV_SIGNER_MODE] === "true";
};
