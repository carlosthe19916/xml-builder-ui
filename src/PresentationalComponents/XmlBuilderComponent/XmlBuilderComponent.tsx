import React from "react";
import { isAppInSignerMode } from "../../Utilities/EnvUtils";

interface Props {
  whenSignerMode: React.ComponentType<any>;
  elseMode?: React.ComponentType<any>;
}

interface State {}

class XmlBuilderComponent extends React.Component<Props, State> {
  render() {
    const {
      children,
      whenSignerMode: SignerComponent,
      elseMode: NoSignerComponent
    } = this.props;

    if (isAppInSignerMode()) {
      return <SignerComponent>{children}</SignerComponent>;
    } else {
      if (NoSignerComponent) {
        return <NoSignerComponent>{children}</NoSignerComponent>;
      } else {
        return <React.Fragment>{children}</React.Fragment>;
      }
    }
  }
}

export default XmlBuilderComponent;
