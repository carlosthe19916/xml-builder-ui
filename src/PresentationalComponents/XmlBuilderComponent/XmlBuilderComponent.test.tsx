import React from "react";
import { shallow } from "enzyme";
import XmlBuilderComponent from "./XmlBuilderComponent";
import { Button } from "@patternfly/react-core";
import { ENV_SIGNER_MODE } from "../../Utilities/EnvUtils";

class SignerModeComponent extends React.Component<{}, {}> {
  render() {
    return <p>Hello</p>;
  }
}

class NoSignerModeComponent extends React.Component<{}, {}> {
  render() {
    return <p>Hola</p>;
  }
}

afterEach(() => {
  delete process.env[ENV_SIGNER_MODE];
});

it("[ENV_SIGNER_MODE=true] Render component", () => {
  process.env = Object.assign(process.env, { [ENV_SIGNER_MODE]: "true" });

  const wrapper = shallow(
    <XmlBuilderComponent whenSignerMode={SignerModeComponent} />
  );
  expect(wrapper).toMatchSnapshot();
});

it("[ENV_SIGNER_MODE=true] Render component with children", () => {
  process.env = Object.assign(process.env, { [ENV_SIGNER_MODE]: "true" });

  const wrapper = shallow(
    <XmlBuilderComponent whenSignerMode={SignerModeComponent}>
      <p>Component children</p>
    </XmlBuilderComponent>
  );
  expect(wrapper).toMatchSnapshot();
});

it("[ENV_SIGNER_MODE=false] Render ELSE component", () => {
  process.env = Object.assign(process.env, { [ENV_SIGNER_MODE]: "false" });

  const wrapper = shallow(
    <XmlBuilderComponent
      whenSignerMode={SignerModeComponent}
      elseMode={NoSignerModeComponent}
    >
      <p>Component children</p>
    </XmlBuilderComponent>
  );
  expect(wrapper).toMatchSnapshot();
});

it("[ENV_SIGNER_MODE=false] Render ELSE empty component", () => {
  process.env = Object.assign(process.env, { [ENV_SIGNER_MODE]: "false" });

  const wrapper = shallow(
    <XmlBuilderComponent whenSignerMode={SignerModeComponent}>
      <p>Component children</p>
    </XmlBuilderComponent>
  );
  expect(wrapper).toMatchSnapshot();
});
