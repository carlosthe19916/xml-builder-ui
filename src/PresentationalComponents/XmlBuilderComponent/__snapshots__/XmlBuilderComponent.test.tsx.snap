// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ENV_SIGNER_MODE=false] Render ELSE component 1`] = `
<NoSignerModeComponent>
  <p>
    Component children
  </p>
</NoSignerModeComponent>
`;

exports[`[ENV_SIGNER_MODE=false] Render ELSE empty component 1`] = `
<Fragment>
  <p>
    Component children
  </p>
</Fragment>
`;

exports[`[ENV_SIGNER_MODE=true] Render component 1`] = `<SignerModeComponent />`;

exports[`[ENV_SIGNER_MODE=true] Render component with children 1`] = `
<SignerModeComponent>
  <p>
    Component children
  </p>
</SignerModeComponent>
`;
